docs/user/openstack/install_upi.md

```diff
@@ -0,0 +1,806 @@
+# OpenStack UPI
+
+
+## Before You Begin
+
+You need these things available:
+
+- OpenStack account with `clouds.yaml` (we'll be using a cloud called `openstack` in this document)
+- Nova flavors (we'll use `m1.xlarge` for masters and `m1.large` for workers)
+- An external subnet you want to use for floating IP addresses (we'll use `external`)
+- The `openshift-install` binary in your `$PATH`
+- The Red Hat CoreOS image in Glance (we'll use `rhcos`)
+- A subnet range for the Nova servers / OpenShift Nodes
+  - can be anything but mustn't conflict with your existing own network
+  - we'll use `192.0.2.0/24`
+- A cluster name you will want to use (we'll use `openshift`)
+- A base domain (we'll use `example.com`)
+- OpenShift Pull Secret
+- DNS infrastructure you can configure (you'll need to add two records there)
+
+The OpenShift API URL will be generated from the cluster name and base domain. E.g.: `https://api.openshift.example.com:6443/`
+
+
+You can validate most of the above by running the following commands:
+
+```sh
+$ export OS_CLOUD=openstack
+$ openstack image show rhcos
+$ openstack network show external
+$ openstack flavor show m1.xlarge
+$ openstack flavor show m1.large
+$ which openshift-install
+```
+
+They should all succeed.
+
+
+## OpenShift Configuration Directory
+
+All the configuration files, logs and installation state is kept in a single directory:
+
+```sh
+$ mkdir -p openstack-upi
+$ cd openstack-upi
+```
+
+## API and Ingress Floating IP Addresses
+
+```sh
+$ openstack floating ip create --description "OpenShift API" <external>
+=> 203.0.113.23
+$ openstack floating ip create --description "OpenShift Ingress" <external>
+=> 203.0.113.19
+```
+
+**NOTE**: throughout this document, we will use `203.0.113.23` as the public IP address for the OpenShift API endpoint and `203.0.113.19` as the public IP for the ingress (`*.apps`) endpoint.
```

Comment:
Nice, I see that you're using the IP address range for documentation as specified in RFC 5737 :+1:

Reply:
I'd been wondering for a while if there's something like example.com, but for TLDs and IP addresses. Turns out there is \o/

Reply:
I also think that adherence to IETF RFCs is of unquestionable value.

However, while our readers will immediately identify `example.com` as a placeholder, on the other hand `192.0.2.0/24` might be copy-pasted in a production setting.

In order to avoid confusion, I propose to be more explicit about the fact that the IP ranges we use in the examples (192.0.2.0/24 and 203.0.113.0/24, aka TEST-NET-1 and TEST-NET-3) are indeed example IPs and must not be used.

From [RFC 5737](https://tools.ietf.org/html/rfc5737):
> Network operators SHOULD add these address blocks to the list of non-routeable address spaces, and if packet filters are deployed, then this address block SHOULD be added to packet filters.

Reply:
Yeah that's a fair point. I'd still like to use the ranges that are actually designed for that purpose rather than actual real ranges that are potentially in use.

That said, adding a warning and making it clear you should pick your own safe range is a completely sensible thing to do.